lib/utils.ts

```diff
@@ -1,10 +1,228 @@
 import crypto from "crypto";
 import { LogLine } from "../types/log";
+import { TextAnnotation } from "../types/textannotation";
 
 export function generateId(operation: string) {
   return crypto.createHash("sha256").update(operation).digest("hex");
 }
 
+export function formatText(textAnnotations: TextAnnotation[]): string {
```

Comment:
General theme here -- appreciate the comments you do have, but I think a more generalized docstring that outlines the process you're taking and why, and then linking the comments back to steps in said docstring would be super helpful here. Right now, I understand that this works better, but if I had to debug this I'd be super lost -- what is a canvas? What are the widths you're calculating? What are the different lines you're referencing?